cypress/e2e/requests/cancel-request.cy.js

```diff
@@ -0,0 +1,187 @@
+          expandAll: true,
+          query: `"id"=="${instanceData.instanceId}"`,
+        }).then((instance) => {
+          instanceData.instanceHRID = instance.hrid;
+        });
+        itemData = testData.folioInstances[0];
+      });
+      RequestPolicy.createViaApi(requestPolicyBody);
+      CirculationRules.getViaApi().then((circulationRule) => {
+        originalCirculationRules = circulationRule.rulesAsText;
+        const ruleProps = CirculationRules.getRuleProps(circulationRule.rulesAsText);
+        ruleProps.r = requestPolicyBody.id;
+        addedCirculationRule =
+          't ' +
+          testData.loanTypeId +
+          ': i ' +
+          ruleProps.i +
+          ' l ' +
+          ruleProps.l +
+          ' r ' +
+          ruleProps.r +
+          ' o ' +
+          ruleProps.o +
+          ' n ' +
+          ruleProps.n;
+        CirculationRules.addRuleViaApi(
+          originalCirculationRules,
+          ruleProps,
+          't ',
+          testData.loanTypeId,
+        );
+      });
+      cy.createTempUser([Permissions.requestsAll.gui, Permissions.checkinAll.gui])
+        .then((userProperties) => {
+          testData.user = userProperties;
+        })
+        .then(() => {
+          UserEdit.addServicePointsViaApi(
+            [testData.servicePoint.id],
+            testData.user.userId,
+            testData.servicePoint.id,
+          );
+
+          cy.login(testData.user.username, testData.user.password, {
+            path: TopMenu.requestsPath,
+            waiter: Requests.waitLoading,
+          });
+          NewRequest.openNewRequestPane();
+          NewRequest.waitLoadingNewRequestPage();
+          NewRequest.enterHridInfo(instanceData.instanceHRID);
+          NewRequest.verifyHridInformation([instanceData.title]);
+          NewRequest.enterRequesterInfoWithRequestType(
+            {
+              requesterBarcode: testData.user.barcode,
+              pickupServicePoint: testData.servicePoint.name,
+            },
+            testData.requestType,
+          );
+          NewRequest.verifyRequestInformation(testData.requestType);
+          NewRequest.saveRequestAndClose();
+          cy.intercept('POST', 'circulation/requests').as('createRequest');
+          cy.wait('@createRequest').then((intercept) => {
+            testData.requestId = intercept.response.body.id;
+            cy.location('pathname').should('eq', `/requests/view/${testData.requestId}`);
+          });
+          NewRequest.verifyRequestSuccessfullyCreated(testData.user.username);
+          RequestDetail.checkRequestStatus('Open - Not yet filled');
+          RequestDetail.checkItemStatus('Paged');
+        });
+    });
+  });
+
+  after('delete test data', () => {
+    cy.getAdminToken().then(() => {
+      RequestPolicy.deleteViaApi(requestPolicyBody.id);
+      Requests.deleteRequestViaApi(testData.requestId);
+      UserEdit.changeServicePointPreferenceViaApi(testData.user.userId, [testData.servicePoint.id]);
```

Comment:
Do you need change service point, if you delete user anyway?

Reply:
Yes, we have to change the service point, because when we can't delete a service point that still in use